types/mydecimal.go

```diff
@@ -1301,6 +1302,12 @@ func (d *MyDecimal) ToHashKey() ([]byte, error) {
 		// thus ErrTruncated may be raised, we can ignore it here.
 		err = nil
 	}
+	precBuff := make([]byte, 4)
```

Comment:
Can you use some fuzz test to check whether `a eq b ==> hash(a) eq hash(b)` always be true?

TiKV's [Decimal hash](https://github.com/tikv/tikv/pull/6047) implementation passed such fuzz test.

Reply:
Here are the tests: https://github.com/lzmhhh123/tidb/blob/6320bea24d3f790714908844cd519e5267c4ac2e/types/mydecimal_test.go#L154, actually, is `a eq b <=> hash(a) eq hash(b)` rather than `a eq b ==> hash(a) eq hash(b)`

Reply:
Fuzz tests are tests that uses a fuzzer to run, instead of unit tests. (Also i don't quite understand the necessarity of ensuring that `Hash(a) eq Hash(b) => a eq b`. If this can be ensured, then there would be no hash collision?)

Reply:
IMHO, if we add fuzzer test, other types of hash functions also need to be involved in fuzzer test. It's another thing. Perhaps we can discuss it in another issue. And for `Hash(a) eq Hash(b) => a eq b`, because we use the function to compare decimal in some places.

Reply:
Agree with @lzmhhh123 , `Hash(a) eq Hash(b) => a eq b` is still necessary, and this should be no hash collision by the unlimited hash size.